### ISSUE TITLE
Issues with interpreter (rename) and ecoreref

### ISSUE THREAD

**Issue by LorenzoBettini** [LorenzoBettini/edelta#93]
This modification works:

```
modifyEcore modifyNet epackage petrinet {
	ecoreref(Net).name = "Petrinet"
	ecoreref(places).lowerBound=1
	ecoreref(transitions).lowerBound=1
}
```

but this one fails

```
modifyEcore modifyNet epackage petrinet {
	ecoreref(Net).name = "Petrinet"
	ecoreref(Petrinet.places).lowerBound=1
	ecoreref(Petrinet.transitions).lowerBound=1
}
```

with error `The method lowerBound(int) is undefined for the type ENamedElement`

**Comment by LorenzoBettini** [comment]
The working version leads to a NPE when running the Java code

**Comment by LorenzoBettini** [comment]
This one works at runtime

```
modifyEcore introduceAbstractArc epackage petrinet {
	addNewEClass("Arc") [
		abstract = true
		val List<EAttribute> attributes = #[ecoreref(PTArc.weight), ecoreref(TPArc.weight)]
//		refactorings.extractSuperclass(it, attributes);
	]
}
```

generating

```java
final List<EAttribute> attributes = ...newArrayList(
getEAttribute("petrinet", "PTArc", "weight"), getEAttribute("petrinet", "TPArc", "weight")));
```

while this one

```
modifyEcore introduceAbstractArc epackage petrinet {
	addNewEClass("Arc") [
		abstract = true
		val List<EAttribute> attributes = #[ecoreref(PTArc.weight), ecoreref(TPArc.weight)]
		refactorings.extractSuperclass(it, attributes);
	]
}
```

leads to wrong generated code

```java
final List<EAttribute> attributes = ...newArrayList(
getEAttribute("petrinet", "Arc", "weight"), getEAttribute("", "", "weight")));
```

Since ```refactorings.extractSuperclass(it, attributes)``` lifts the attributes in the superclass

So the interpreter breaks code generation

**Comment by LorenzoBettini** [comment]
`ecoreref(Petrinet.places).lowerBound` is actually resolved (you can jump to `setLowerBound` in the editor); that's probably the type system that previously detected the error and it does not remove it later when everything is resolved.

This should be already handled by `EdeltaLinkingDiagnosticMessageProvider`?

**Comment by LorenzoBettini** [comment]
Indeed this one works without errors:

```
println(ecoreref(Petrinet.places).lowerBound)
```

**Comment by LorenzoBettini** [comment]
It works also with setter

```
ecoreref(Petrinet.places).setLowerBound(1)
```

So it's just the Xbase syntactic sugar syntax?

**Comment by LorenzoBettini** [comment]
Looks like it's only syntactic sugar syntax:

```
	@Test
	def void testReferenceToAttributeofRenamedClassInModifyEcore() {
		'''
		import org.eclipse.emf.ecore.EClass

		metamodel "foo"

		modifyEcore aTest epackage foo {
			ecoreref(foo.FooClass).name = "RenamedClass"
			ecoreref(foo.RenamedClass).setAbstract(true)
			ecoreref(foo.RenamedClass).abstract = true
		}
		'''.parseWithTestEcore.assertNoErrors
	}
```

`setAbstract` is resolved, the `=` is not.

**Comment by LorenzoBettini** [comment]
Because that's an `XAssignment` not an `XMemberFeatureCall` (while we do `if (contextObject instanceof XMemberFeatureCall)`)

**Comment by LorenzoBettini** [comment]
This is still an issue:

```
	// you can rename existing classes
	ecoreref(MyOtherEClass).name = "RenamedClass"
	// and the renamed version is immediately available
	ecoreref(RenamedClass) => [abstract = true]
```

`abstract` has an error marker but it is resolved.

Probably we need a more general approach that cleans the errors generated by the type system when we clean the cache during the interpretation? See also #97

**Comment by LorenzoBettini** [comment]
Closed by #99
